sessionctx/variable/sysvar.go

```diff
@@ -490,7 +496,7 @@ var defaultSysVars = []*SysVar{
 	{Scope: ScopeGlobal | ScopeSession, Name: CharacterSetClient, Value: mysql.DefaultCharset, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
 		return checkCharacterValid(normalizedValue, CharacterSetClient)
 	}},
-	{Scope: ScopeNone, Name: Port, Value: "4000"},
+	{Scope: ScopeNone, Name: Port, Value: "4000", Type: TypeUnsigned},
```

Comment:
I've added a `MinValue` and `MaxValue`. I used values [identical to MySQL](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_port) to be compatible. Note that the minimum value is 0 instead of 1 which is [interesting](https://www.lifewire.com/port-0-in-tcp-and-udp-818145).